### PR TITLE
Change XY formatting on LCD

### PR DESCRIPTION
**Pull request by thinkyhead**
According to #123 negative values for XY at or below -100 are displaying incorrectly, dropping the first digit. Deltas can easily have XY values in this range. This PR adds a function to display floats/ints formatted like `_123`, `-123`, `_-12`, or `__-1` as appropriate and applies it to the XY coordinates on Hitachi displays. It also moves the Z value to the right to be consistent with the XY formatting.

### CODE REVIEW

**Comment by AnHardt** [comment]
A minor enhancement for the documentation.
```
16x4   |000/000 B000/000|
       |SD100%  Z 000.00| <
       |F100%     T--:--|
       |0123456789012345|

**Comment by thinkyhead** [comment]
Rebased. I'll probably submit this fix to the RCBugFix branch as well.

File: ultralcd.h [hunk 1]
```diff
@@ -116,21 +116,23 @@
...
-char* ftostr52(const float& x);
+char* itostr2(const uint8_t &x);
+char* itostr31(const int &xx);
+char* itostr3(const int &xx);
+char* itostr3left(const int &xx);
+char* itostr4(const int &xx);
+char* itostr4sign(const int &x);
+
+char* ftostr3(const float &x);
+char* ftostr4sign(const int &x);
```
Comment (AnHardt) [inline ultralcd.h]: int ??? --> float !!!

File: ultralcd.cpp [hunk 2]
```diff
@@ -2086,6 +2087,30 @@ char* itostr4(const int& xx) {
   return conv;
 }
 
+// Convert int to rj string with _123, -123, _-12, or __-1 format
+char *itostr4sign(const int &x) {
+  long xx = abs(x);
```
Comment (AnHardt) [inline ultralcd.cpp]: int is save here for 4 digits.

File: ultralcd.cpp [hunk 3]
```diff
@@ -1901,10 +1901,11 @@ void lcd_reset_alert_level() { lcd_status_message_level = 0; }
...
 
-// Convert float to string with +123.4 format
-char* ftostr3(const float& x) {
-  return itostr3((int)x);
-}
+// Convert float to rj string with 123 or -12 format
+char *ftostr3(const float &x) { return itostr3((int)x); }
+
+// Convert float to rj string with _123, -123, _-12, or __-1 format
+char *ftostr4sign(const int &x) { return itostr4sign((int)x); }
```
Comment (AnHardt) [inline ultralcd.cpp]: ftostr4sign(const int &x)
float !

**Comment by AnHardt** [comment]
Replaced by #265

**Pull request closed without merging**